typing/printtyp.ml

```diff
@@ -2432,19 +2432,20 @@ let explain mis ppf =
   | None -> ()
   | Some explain -> explain ppf
 
-let warn_on_missing_def env ppf t =
+type env_kind = Normal_env | Abstract_env of string
```

Comment:
I don't know if this is a policy yet, but I concur that messages at destination to humans should be emitted in one piece if possible, in order to have the full context available.

I see three major benefits for this constraint on the error message:
* it can be read and copy edited in one place
* it can be formatted as a whole
* it can be translated with enough context to enable a good translation